core/lib/src/form/error.rs

```diff
@@ -804,9 +804,9 @@ impl fmt::Display for ErrorKind<'_> {
                 }
             }
             ErrorKind::InvalidChoice { choices } => {
-                match choices.as_ref() {
-                    &[] => write!(f, "invalid choice")?,
-                    &[ref choice] => write!(f, "expected {}", choice)?,
+                match *choices.as_ref() {
+                    [] => write!(f, "invalid choice")?,
+                    [ref choice] => write!(f, "expected {}", choice)?,
```

Comment:
I find the previous version easier to read. Let's use that one.

Reply:
I've added the following allow :
`#[allow(clippy::match_ref_pats)] // It's more readable to repeat the reference.`
As I said earlier, it will be fine once we get use this way of writing code. So I would recommend respecting Clippy's style warnings because they are present to make the code more standard between projects.

Reply:
I'm happily used to both versions but prefer the former. This one adds a deref of a type that is almost always behind a reference, which is odd. Let's stick to the previous version.

Reply:
I'm ok with that. I think this comment will be more inline with the given explanation :
```
#[allow(clippy::match_ref_pats)] // Cow is almost always behind a reference, we'll keep it that way for consistency.
```